app/src/main/java/com/duckduckgo/app/browser/webview/MaliciousSiteBlockerWebViewIntegration.kt

```diff
@@ -197,9 +242,12 @@ class RealMaliciousSiteBlockerWebViewIntegration @Inject constructor(
         processedUrls.clear()
     }
 
-    override fun onSiteExempted(url: Uri) {
+    override fun onSiteExempted(
+        url: Uri,
+        feed: Feed,
+    ) {
         val convertedUrl = URLDecoder.decode(url.toString(), "UTF-8").lowercase()
-        exemptedUrlsHolder.exemptedMaliciousUrls.add(convertedUrl)
+        exemptedUrlsHolder.exemptedMaliciousUrls.add(ExemptedUrl(convertedUrl.toUri(), feed))
```

Comment:
why receiving an url: Uri -> decoding to Url -> and converting to Uri again?

Reply:
we need to decode so we can match 2 URLs that are technically the same, but encoded differently. We could technically then store it as String, but after my experience working in performance, I prefer to use Uri whenever possible, and only rely on strings when necessary, as it's the case here